core/src/erasure.rs

```diff
@@ -435,14 +426,16 @@ pub mod test {
             data_blobs[i].write().unwrap().set_slot(1);
         }
 
-        let coding_blobs = coding_generator.next(&data_blobs[0..NUM_DATA - 1]);
-        assert_eq!(coding_blobs.len(), 0);
+        let coding_blobs = session.generate_coding_shared(&data_blobs[0..NUM_DATA - 1]);
+        assert!(coding_blobs.is_err());
 
-        let coding_blobs = coding_generator.next(&data_blobs[NUM_DATA..]);
+        let coding_blobs = session
+            .generate_coding_shared(&data_blobs[NUM_DATA..])
+            .unwrap();
```

Comment:
much simpler to deal with no coding blobs than to have to if let ....

let's back out any generator-side changes, please

Reply:
Yes I'll do that. I didn't notice that's how broadcast was using the generator, I guess because I just saw it returning Vec and didn't consider that it could be empty according to the old impl.

I'll put the generator back in and change it to store a session as well.